### PR TITLE
DOC: Document 1.14.4 release.

### PR DESCRIPTION
**Pull request by charris**
[ci skip]

### CODE REVIEW

File: doc/source/release.rst [hunk 1]
```diff
@@ -2,6 +2,8 @@
 Release Notes
 *************
 
+.. include:: ../release/1.14.4-notes.rst
+.. include:: ../release/1.14.3-notes.rst
```
Comment (ahaldane) [inline doc/source/release.rst]: Ah, I missed this in 1.14.3. Shall I add a paragraph to the release walkthrough about which release/contributor files should be updated?

Comment (charris) [reply doc/source/release.rst]: Sure. I'm too close to the process to see those things and anything that helps the next person to make releases is good.

File: doc/release/1.14.4-notes.rst [hunk 2]
```diff
@@ -2,17 +2,58 @@
...
+
+There are also improvements to printing of long doubles on PPC platforms. All
+is not yet perfect on that platform, the whitespace padding is still incorrect
+and is to be fixed in numpy 1.15, consequently NumPy still fails some
+printing-related (and other) unit tests on ppc systems. However, the printed
+values are now correct.
+
+Note that NumPy will warn on import if it detects the use of PyQt5 or other
+modules that have resulting in incorrect float32 `dot` results on the MAC when
+working in the Anaconda enviroment due to a subtle interaction with MVK. This
```
Comment (mattip) [inline doc/release/1.14.4-notes.rst]: MVK -> MKL

File: doc/release/1.14.4-notes.rst [hunk 3]
```diff
@@ -2,17 +2,58 @@
...
+* fixes for bus faults on SPARC and older ARM due to incorrect alignment
+  checks.
+
+There are also improvements to printing of long doubles on PPC platforms. All
+is not yet perfect on that platform, the whitespace padding is still incorrect
+and is to be fixed in numpy 1.15, consequently NumPy still fails some
+printing-related (and other) unit tests on ppc systems. However, the printed
+values are now correct.
+
+Note that NumPy will warn on import if it detects the use of PyQt5 or other
```
Comment (pv) [inline doc/release/1.14.4-notes.rst]: warn -> error

**Pull request merged**